Src/AwesomeAssertions/Primitives/StringEndStrategy.cs

```diff
@@ -19,24 +20,70 @@ public StringEndStrategy(IEqualityComparer<string> comparer, string predicateDes
 
     public void ValidateAgainstMismatch(AssertionChain assertionChain, string subject, string expected)
     {
-        assertionChain
-            .ForCondition(subject!.Length >= expected.Length)
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but {{1}} is too short.", expected, subject);
-
-        if (!assertionChain.Succeeded)
+        var (mismatchInSubject, mismatchInExpectation) = IndexOfLastMismatch(subject, expected, comparer);
+        if (mismatchInExpectation < 0)
         {
             return;
```

Comment:
Please, add a comment explaining why we can leave the method without further action.